### PR TITLE
Get local rule descriptions for Hotspots

### LINKED ISSUE #4268: Get local rule descriptions for  Hotspots

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
fixes #4268

### CODE REVIEW

File: src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs [hunk 1]
```diff
@@ -133,15 +131,17 @@ private static bool ProcessResource(string fullResourceName)
...
-                var doc = testSubject.Create(jsonRuleInfo);
-
-                // Quick sanity check that something was produced
-                // Note: this is a quick way of getting the size of the document. Serializing the doc to a string
-                // and checking the length takes much longer (around 25 seconds)
-                var docLength = doc.ContentStart.DocumentStart.GetOffsetToPosition(doc.ContentEnd.DocumentEnd);
-                Console.WriteLine($"{jsonRuleInfo.FullRuleKey}: size = {docLength}");
-                docLength.Should().BeGreaterThan(30);
-
+                if (!string.IsNullOrWhiteSpace(jsonRuleInfo.Description))
```
Comment (ugras-ergun-sonarsource) [inline src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs:134]: some hotspot rules do not have the description field. (they have descriptionsections filled)

File: src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs [hunk 2]
```diff
@@ -157,17 +157,5 @@ private static string ReadResource(string fullResourceName)
             using var stream = new StreamReader(ResourceAssembly.GetManifestResourceStream(fullResourceName));
             return stream.ReadToEnd();
         }
-
-        private static SonarCompositeRuleId GetCompositeRuleIdFromResourceName(string fullResourceName)
```
Comment (ugras-ergun-sonarsource) [inline src/Education.UnitTests/SimpleRuleHelpXamlBuilderTests.cs:161]: this method was orphaned so I've removed it.

File: src/Rules.UnitTests/LocalRuleMetadataProviderTests.cs [hunk 3]
```diff
@@ -94,7 +94,6 @@ void CheckRule(string fullResourceName)
 
                 actual.FullRuleKey.Should().Be(ruleId.ToString());
                 actual.Description.Should().Be(expectedDescription);
-                actual.Description.Should().NotBeNullOrWhiteSpace();
```
Comment (ugras-ergun-sonarsource) [inline src/Rules.UnitTests/LocalRuleMetadataProviderTests.cs:97]: same thing about the empty description field.

**Comment by ugras-ergun-sonarsource** [comment]
I had rebuilt the solution and it created the individual files for hotspots. From what I understand from the code it should now work without any more changes.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**